core/templates/dev/head/pages/library/Library.js

```diff
@@ -80,6 +80,47 @@ oppia.controller('Library', [
       $http.get('/libraryindexhandler').success(function(data) {
         $scope.libraryGroups = data.activity_summary_dicts_by_category;
 
+        $scope.userIsLoggedIn = GLOBALS.userIsLoggedIn;
+        $scope.activitiesOwned = {explorations: {}, collections: {}};
+        if ($scope.userIsLoggedIn) {
+          $http.get('/creatordashboardhandler/data')
+            .then(function(response) {
+              $scope.libraryGroups.forEach(function(libraryGroup) {
+                var activitySummaryDicts = libraryGroup.activity_summary_dicts;
+
+                var ACTIVITY_TYPE_EXPLORATION = 'exploration';
+                var ACTIVITY_TYPE_COLLECTION = 'collection';
+                activitySummaryDicts.forEach(function(activitySummaryDict) {
+                  if (activitySummaryDict.activity_type === (
+                    ACTIVITY_TYPE_EXPLORATION)) {
+                    $scope.activitiesOwned.explorations[
+                      activitySummaryDict.id] = false;
+                  } else if (activitySummaryDict.activity_type === (
+                    ACTIVITY_TYPE_COLLECTION)) {
+                    $scope.activitiesOwned.collections[
+                      activitySummaryDict.id] = false;
+                  } else {
+                    AlertsService.clearWarnings();
+                    AlertsService.addWarning(
+                      'Error encountered in activity type.');
```

Comment:
Too vague. Be specific about the error: the message should convey that the activity type is wrong (and maybe display it).

Reply:
Got it! Thanks!

Reply:
How about 
$log.error('INVALID ACTIVITY TYPE: Activity(id: ' + activitySummaryDict.id + ') has invalid activity type.')?

Reply:
Well, users won't see $log.error.

The new message seems good. Perhaps also print the activity type?